tests/integration/targets/ec2_vpc_dhcp_option/tasks/main.yml

```
@@ -261,17 +257,18 @@
       delete_old: False
     register: dhcp_options
 
+  - debug:
+      msg: '{{ dhcp_options.new_options.keys() }}'
+
   - assert:
       that:
         - dhcp_options.changed
         - dhcp_options.new_options
         # FIXME extra keys are returned unpredictably
-        - dhcp_options.new_options.keys() | list | sort is superset(['netbios-name-servers', 'netbios-node-type', 'ntp-servers'])
+        - dhcp_options.new_options.keys() | list | sort == ['netbios-name-servers', 'netbios-node-type', 'ntp-servers']
```

```suggestion
        - dhcp_options.new_options.keys() | list | sort == ['netbios-name-servers', 'netbios-node-type', 'ntp-servers']
```

I was thinking that maybe failing if a new option was added is the best option here.  However, I think if someone wants to add additional info, we should probably get them to return the boto3 style data alongside the original boto2 data.

Maybe worth a comment here about why we fail if *extra* return values are passed?

---

Sorry, I may just be being dense but I'm not sure what you're asking on this.  We should only have these 3 keys here because those should be the only 3 keys that are configured in this option set's DhcpConfiguration at this point.

Are you asking if we should return the `normalized_config()` (which attempts to preserve `new_config` from the original module insofar as `new_config` wasn't even entirely consistent in the first place) as well as just dumping the boto3 response?

---

> We should only have these 3 keys here because those should be the only 3 keys that are configured in this option set's DhcpConfiguration at this point.

Ah, ok so it's **supposed** to have done something approximating `purge_none_params`

Most of the time we do `- '"foo" in bar.baz'` to say we only care that the mentioned keys are there, not that it must be **only** specific keys (it gives us flexibility if Amazon add extra features).  I'm worried about non-standard examples knocking about that'll confuse people.  I'd be good with just a comment that "this is non-standard behaviour because..."

> Are you asking if we should return the `normalized_config()` (which attempts to preserve `new_config` from the original module insofar as `new_config` wasn't even entirely consistent in the first place) as well as just dumping the boto3 response?

In general, Yes.

---

Gotcha.  Added a quick take at that, no tests yet (in the current incarnation it would solve the need to return tags at least).  I really don't like the whole Key, Values, Value structure that amazon is giving us here but that's what we have to work with.